### PR TITLE
Responsive redesign for header.tpl.html to fit for mobile devices.

### PR DESCRIPTION
**Pull request by sushanthkannan**
Made changes to solve the overlapping issue in header for mobile device view and medium devices so that it is responsive.I have uploaded the screenshot of existing mobile view and the view after making changes. The ones in the laptop chrome browser is the view after making the changes and the mobile screenshot is existing mobile view.
Screen Shot 2019-08-23 at 4 48 22 pm 
existing

Screen Shot 2019-08-23 at 4 45 58 pm
Screen Shot 2019-08-23 at 4 46 42 pm
Screen Shot 2019-08-23 at 4 47 36 pm
Screen Shot 2019-08-23 at 4 47 08 pm

.

### CODE REVIEW

File: src/app/common/header/header.tpl.html [hunk 1]
```diff
@@ -30,6 +38,8 @@
             <unit-dropdown></unit-dropdown>
           </div>
         </li><!--/selected-unit-->
+            </div>
+            <div class="col-md-4" style="padding-left:0px;">
```
Comment (jakerenzella) [inline src/app/common/header/header.tpl.html]: We don't want inline-styling

File: src/app/common/header/header.tpl.html [hunk 2]
```diff
@@ -10,14 +10,22 @@
...
+          <div class="row">
+            <div class="col-md-1" style="padding-left:0px;padding-right:50px;">
         <li>
           <a class="home" ui-sref="home">
             <i class="logo"></i>
           </a>
         </li><!--/home-->
-        <li>
+            </div>
+        <div class="col-md-6" style="padding-right:100px;padding-left:0px;">
```
Comment (jakerenzella) [inline src/app/common/header/header.tpl.html]: inline-styling

File: src/app/common/header/header.tpl.html [hunk 3]
```diff
@@ -10,14 +10,22 @@
...
   <!-- Collect the nav links, forms, and other content for toggling -->
   <div class="collapse navbar-collapse" ng-class="!navCollapsed && 'in'" ng-show="!navCollapsed">
+    <div class="container" style="width:100%">
+      <div class="row">
+        <div class="col-md-6">
     <ul class="nav navbar-nav navbar-left">
       <ul class="breadcrumb">
+        <div class="container" style="width:100%">
+          <div class="row">
+            <div class="col-md-1" style="padding-left:0px;padding-right:50px;">
```
Comment (jakerenzella) [inline src/app/common/header/header.tpl.html]: inline-styling

**Comment by sushanthkannan** [comment]
Hey @jakerenzella  i have made the changes and did the styling in the stylesheet. Also fixed the overlap issue in the desktop view.
Desktop view
Mobile view
Laptop view
tabletview

**Comment by jakerenzella** [comment]
@sushanthkannan What happened to the `/` characters between the breadcrumb items on Desktop? I am worried that the mobile fixes are affecting the desktop view, potentially in other ways too, which shouldn't be the case.

I think this needs to be discussed more in general. What you have is better than the current behavior for mobile, but it still a patch. We probably shouldn't have a breadcrumb at all on mobile, it doesn't make much sense.

For example, why is the unit and view dropdowns below the logo? This is taking up a fair amount of height.

I think for a good solution to this:

* Keep desktop the same
* For mobile, small logo (currently present)
* Remove unit switcher on mobile only, they can go home via logo and then switch units.
* Single dropdown item for the task switching (which needs to display correctly when expanded)

File: src/app/common/header/header.tpl.html [hunk 4]
```diff
@@ -10,14 +10,22 @@
...
   <div class="collapse navbar-collapse" ng-class="!navCollapsed && 'in'" ng-show="!navCollapsed">
+    <div class="container" id="header-container">
+      <div class="row">
+        <div class="col-md-6">
     <ul class="nav navbar-nav navbar-left">
       <ul class="breadcrumb">
+        <div class="container" id="breadcrumb-container">
+          <div class="row">
+            <div class="col-md-1" id="breadcrumb-col1">
         <li>
```
Comment (jakerenzella) [inline src/app/common/header/header.tpl.html]: Indentation here looks incorrect

File: src/app/common/header/header.tpl.html [hunk 5]
```diff
@@ -10,14 +10,22 @@
...
   <!-- Collect the nav links, forms, and other content for toggling -->
   <div class="collapse navbar-collapse" ng-class="!navCollapsed && 'in'" ng-show="!navCollapsed">
+    <div class="container" id="header-container">
+      <div class="row">
+        <div class="col-md-6">
     <ul class="nav navbar-nav navbar-left">
       <ul class="breadcrumb">
+        <div class="container" id="breadcrumb-container">
+          <div class="row">
+            <div class="col-md-1" id="breadcrumb-col1">
```
Comment (jakerenzella) [inline src/app/common/header/header.tpl.html]: Why does each column need a unique ID?

Comment (sushanthkannan) [reply src/app/common/header/header.tpl.html]: I have given each column a unique id as I have given different padding for the columns.

**Comment by sushanthkannan** [comment]
Hi @jakerenzella I made the changes as requested but couldn't resolve the slash missing issue. I gave a pull request and it went as a separate PR. Please check that.

**Pull request closed without merging**